data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceService.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.IndexParametersConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.model.ServiceInfo;
+import org.opensearch.dataprepper.plugins.source.opensearch.service.ElasticSearchService;
+import org.opensearch.dataprepper.plugins.source.opensearch.service.HostsService;
+import org.opensearch.dataprepper.plugins.source.opensearch.service.OpenSearchService;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.OpenSearchTimerWorker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Timer;
+
+/**
+ *  Service that will call both worker classes
+ */
+public class OpenSearchSourceService {
+    private static final Logger LOG = LoggerFactory.getLogger(OpenSearchSourceService.class);
+
+    private final OpenSearchService openSearchService;
+
+    private final ElasticSearchService elasticSearchService;
+
+    private final HostsService hostsService;
+
+    private final OpenSearchSourceConfiguration sourceConfig;
+
+    private final BufferAccumulator bufferAccumulator;
+
+    private final Timer timer = new Timer();
+
+    public OpenSearchSourceService(final OpenSearchSourceConfiguration sourceConfig,
+                                   final HostsService hostsService,
+                                   final OpenSearchService openSearchService,
+                                   final ElasticSearchService elasticSearchService,
+                                   final BufferAccumulator bufferAccumulator){
+        this.sourceConfig = sourceConfig;
+        this.hostsService = hostsService;
+        this.openSearchService = openSearchService;
+        this.elasticSearchService =elasticSearchService;
+        this.bufferAccumulator = bufferAccumulator;
+    }
+
+    public void processHosts(){
+        sourceConfig.getHosts().forEach(host ->{
```

Comment:
This is not an accurate reflection of the `hosts` parameter.

The `hosts` defines the hosts that make up the OpenSearch cluster. These are all hosts within one cluster. The `opensearch-java` client will handle this for you automatically.

So the `opensearch` source will interact with exactly one OpenSearch cluster. This cluster may have multiple hosts.

You can see the correct usage of `hosts` in the `opensearch` sink where it provides this list of hosts directly to the OpenSearch client as a list. You still get exactly one client from this list of hosts.

https://github.com/opensearch-project/data-prepper/blob/1c6787b5b29239e4847cce13b29a4424feff0a78/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfiguration.java#L244-L250

See this discussion for more details: https://github.com/opensearch-project/data-prepper/pull/2746#issuecomment-1563286326